pkg/eden/qemu.go

```diff
@@ -41,10 +43,40 @@ func StopSWTPM(stateDir string) error {
 	return utils.StopCommandWithPid(pidFile)
 }
 
+func startQMPLogger(qmpSock string, qmpLog string) error {
+	// Wait a bit for a socket to be created
+	n := 5
+	for n > 0 {
+		sinfo, err := os.Stat(qmpSock)
+		if err != nil || sinfo.Mode().Type() != fs.ModeSocket {
+			time.Sleep(1 * time.Second)
+			n--
+			continue
+		}
+		break
+	}
+	if n == 0 {
+		// Give up, probably something went wrong
+		return fmt.Errorf("startQMPLogger: no QMP socket")
+	}
+	shellcmd := fmt.Sprintf(
+		"echo '{\"execute\": \"qmp_capabilities\"}' | " +
+		"socat -t0 -,ignoreeof UNIX-CONNECT:%s > %s",
+		qmpSock, qmpLog)
+	opts := []string{
+		"-c", shellcmd,
+	}
+	if err := utils.RunCommandNohup("sh", "", "", opts...); err != nil {
```

Comment:
Will this command stop when qemu exits?
If not, then we will have to somehow stop it from here: https://github.com/lf-edge/eden/blob/master/pkg/eden/qemu.go#L211-L213

Reply:
Yes. socat dies when socket is closed.